### PR TITLE
Stats: Fix empty modules in locales with alternate date formats

### PR DESCRIPTION
**Pull request by yoavf**
TL;DR: Make sure we use the 'en' locale when doing period manipulations on a moment date. The returned date is formatted according to current locale, but then later used to access property names which won't work when localized. `Moment.format()` should only be used for **display** purposes.

The issue flows from

```periodRange = rangeOfPeriod( this.options.period, this.options.date ),
startDate = periodRange.startOf.format( 'YYYY-MM-DD' );```

The above doesn't only change the date format, it also localizes it, assuming momentjs is currently loaded with a locale.

That would be ok for display purposes, but then we do ```payload.days[ startDate ]```  which assumes the date in English. If the locale is in Arabic, `startDate` would be something like `٢٠١٦-٠٦-٢٣`.

The solution presented here is a **workaround**, meant as a quick fix. A better solution would be to make sure we're not using `moment.format()` for anything but display purposes.

cc @timmyc who recently worked on this parts :)

**Test:**
1. Change UI locale to Arabic in your user settings
2. Check your stats and notice modules are populated with data in tis branch, but aren't in master.




Test live: https://calypso.live/?branch=fix/non-latin-dates-stats

### CODE REVIEW

**Comment by timmyc** [comment]
Interesting - I thought with moment localization would only happen when calling one of the specific localized strings like `L` and `LL` - but re-reading the docs it appears that assumption is incorrect :)

@yoavf throughout the stats code moment.format is leaned on quite heavily for date manipulation to not only query for the correct data from lists/flux/redux, but also when parsing [[#](https://github.com/Automattic/wp-calypso/blob/master/client/lib/stats/stats-list/stats-parser.js)], [[#](https://github.com/Automattic/wp-calypso/blob/master/client/state/stats/lists/selectors.js#L63)].  It seems if we went with direct calls to [DateTimeFormat](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/DateTimeFormat#Using_DateTimeFormat) we would encounter similar i18n issues.

As such is your work-around acceptable to use in other parts of calypso where we are using dates as keys like this?  If so I'd be happy to add a few more updates to this branch.

**Comment by yoavf** [comment]
> As such is your work-around acceptable to use in other parts of calypso where we are using dates as keys like this? If so I'd be happy to add a few more updates to this branch.

Yes, I'd say it's ok. To clarify, the issue only came up when we [recently](https://github.com/Automattic/i18n-calypso/commit/08466fac7881b7fa897d442ce768ba88047445d0) started including more complete moment.js data - so it's not something that could have easily be spotted before. I agree there's no easy alternative, though it'd be great to make one.

Either way - thanks, and yes - go ahead with any additional updates.

**Comment by timmyc** [comment]
> started including more complete moment.js data

Ah!  Good to know, because when we were first working on stats ( 2 years ago - wow ), I'm fairly certain we tested in various locales to try to find gotchas like this.  I'll get a few more fixes in place and we can get this shipped.

**Comment by timmyc** [comment]
Okay pushed up a few more moment fixes.  The following test suites both pass:

`npm run test-client client/lib/stats/stats-list`
`npm run test-client lib/stats/stats-list `

File: client/state/stats/lists/selectors.js [hunk 1]
```diff
@@ -59,7 +59,7 @@ export const getSiteStatsPostStreakData = createSelector(
 
 		if ( streakData && streakData.data ) {
 			Object.keys( streakData.data ).forEach( ( timestamp ) => {
-				const postDay = i18n.moment.unix( timestamp );
+				const postDay = i18n.moment.unix( timestamp ).locale( 'en' );
```
Comment (gwwar) [inline client/state/stats/lists/selectors.js]: Does moment support all of our locales? We can get the current user locale from the current user in the redux tree:

```javascript
import { getCurrentUserLocale } from 'state/current-user/selector';
getCurrentUserLocale( state );
```

Comment (gwwar) [inline client/state/stats/lists/selectors.js]: Oh sorry, misread the PR description. We'll probably want to add comments for why we're hard coding the locale since these will stick out pretty badly in the code.

File: client/my-sites/stats/stats-video-details/index.jsx [hunk 2]
```diff
@@ -32,7 +32,6 @@ module.exports = React.createClass( {
 			'is-video-details',
 			{
 				'is-loading': isLoading,
-				'is-showing-info': this.state.showInfo,
```
Comment (yoavf) [inline client/my-sites/stats/stats-video-details/index.jsx]: Intentional change?

Comment (timmyc) [inline client/my-sites/stats/stats-video-details/index.jsx]: Yes this was actually a bug I found from an old mixin being removed.

**Comment by yoavf** [comment]
@timmyc with the additional changes, the labels in the graphs are no longer localized:

Up to a2c0114:
screen shot 2016-06-28 at 17 25 58

Current:
screen shot 2016-06-28 at 17 25 48

**Comment by timmyc** [comment]
@yoavf indeed the fixes were a bit deeper then just the stats parser.  In d4670ea logic was added to the stats controller and changes to the original update I performed in the stats-parser to fix the problem you noted above.

In 9c98a92 the "Post Streak" component seen on the insights page was fixed to work with Arabic as well.  The component needs some design love for the RTL and Arabic combo but that is out of scope here.

I have tested out the following, but would be great for one more person to test both with Arabic, and a non-RTL language to make sure all is well:

- View a site stats page, verify numbers show up as expected in the chart tabs.
- Verify the Chart x-axis labels are localized properly
- Verify the correct date bar is highlighted in the chart
- Click on a chart bar, verify that bar is highlighted, and the data updates throughout the page
- Click on a stats summary link ( i.e. Post & Pages View All ) - verify the data looks correct
- View a stats insights page and verify that data is shown in the "Post Trends" visualization

**Comment by yoavf** [comment]
Tested AR/HE/EN_GB - mostly everything works as expected, except that I'm not seeing data in urls such as `stats/month/posts/xxx.wordpress.com?startDate=2015-08-01` in Arabic.

Works in the other locales I've tried.

**Comment by timmyc** [comment]
@yoavf I can't reproduce the issue in Arabic, here is a screen grab after clicking on a date bar in the chart - data showing up in the chart tabs, and in the components below:

stats_ _wordpress_com_news_ _wordpress_com

**Comment by yoavf** [comment]
@timmyc sorry - can't reproduce now either, must has been something in my setup.
:shipit:
